File: Backend/src/main/java/com/naoufalayache/authentication/services/UserService.java
package com.naoufalayache.authentication.services;

import java.util.List;
import java.util.Optional;

import com.naoufalayache.authentication.model.User;

public interface UserService {
    public Optional<User> findById(Long userId);
    public List<User> getUserByEmail(String email, int page, int size);
}

File: Frontend/src/app/services/user.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ConfigService } from './config.service';
import { User } from '../models/model';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  constructor(
    private http: HttpClient,
    private configService: ConfigService,
  ) {}

  getUserByEmail(email: string, page: number | null, size: number) {
    const params = new HttpParams()
      .set('email', email)
      .set('page', page ? page : 0)
      .set('size', size ? size : 20);
    return this.http.get<User[]>(`${this.configService.apiUrl}/user`,{params})
  }
}
